### PR TITLE
added alcohol bereal functionality

### PR DESCRIPTION
**Pull request by aparinrodion**

### CODE REVIEW

**Comment by vladilyushchenko** [comment]
Dear Rodion,

I hope this letter finds you well. I am writing to discuss the merge conflicts that have arisen in our current project. As you are likely aware, these conflicts are causing delays and difficulties in our work. I would greatly appreciate it if you could take some time to resolve these conflicts as soon as possible so that we can move forward with the project.

In addition, I would like to invite you to join me for some shots of vodka on April 8th. It would be a great opportunity for us to unwind and relax after the hard work we have been putting into the project. I believe that getting to know each other outside of work will help us to work more effectively together in the future.

If you are available to meet up on April 8th, please let me know and we can finalize the details. I am looking forward to hearing from you and hopefully, we can resolve the merge conflicts soon and celebrate our hard work with some vodka.

Best regards,

Vlad

**Comment by aparinrodion** [comment]
>

File: src/main/java/com/fbc/bot/bereal/service/AlcoholBerealGenerator.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+        while (firstUserIndex == secondUserIndex) {
+            secondUserIndex = random.nextInt(users.size());
+        }
+
+        return generateText(users.get(firstUserIndex), users.get(secondUserIndex));
+    }
+
+    private String generateText(User firstUser, User secondUser) {
+        return "@" + firstUser.getUserName() + " бухает с " + "@" + secondUser.getUserName();
```
Comment (vladilyushchenko) [inline src/main/java/com/fbc/bot/bereal/service/AlcoholBerealGenerator.java:33]: Add several messages choose them randomly:
1. user1 и user2, бармен вас уже заждался! Быстрее к нему!
3. user1, срочно бери user2 и бегом на бар!
4. user1 и user2, где вас буря носит? Бегом на бар, вы проштрафились.
5. user1 и user2, идите на бар, скажите что от меня - я проставляюсь

File: src/main/java/com/fbc/bot/telegram/service/article/AlcoholFriendInlineResultArticle.java [hunk 2]
```diff
@@ -21,7 +21,7 @@ public InlineQueryResultArticle getArticle(Update update) {
         InputTextMessageContent content = new InputTextMessageContent();
         content.setMessageText("I love vodka");
         article.setInputMessageContent(content);
-        // todo: add pictures
+
```
Comment (vladilyushchenko) [inline src/main/java/com/fbc/bot/telegram/service/article/AlcoholFriendInlineResultArticle.java:24]: return todo

File: src/main/resources/application.yaml [hunk 3]
```diff
@@ -29,5 +29,16 @@ spring:
...
+    mainChatId: -970277358
+
+bereal:
+  enabled: true
+
+feign:
+  client:
+    config:
+      default:
+        connectTimeout: 160000000
```
Comment (vladilyushchenko) [inline src/main/resources/application.yaml:43]: ?

File: src/main/java/com/fbc/bot/bereal/service/AlcoholBerealGenerator.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class AlcoholBerealGenerator {
+
+    private final UserService userService;
+    private final Random random = new Random();
+    private final BerealTextTemplateService textTemplateService;
+    /*private final List<String> textTemplates = List.of("" +
```
Comment (vladilyushchenko) [inline src/main/java/com/fbc/bot/bereal/service/AlcoholBerealGenerator.java:21]: delete

File: src/main/java/com/fbc/bot/bereal/service/BerealTextTemplateService.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+@Service
+@RequiredArgsConstructor
+public class BerealTextTemplateService {
+
+    private final BerealTextTemplateRepository repository;
+
+    public List<BerealTextTemplate> getAll() {
+        List<BerealTextTemplate> textTemplates = new ArrayList<>();
+        repository.findAll().forEach(textTemplates::add);
```
Comment (vladilyushchenko) [inline src/main/java/com/fbc/bot/bereal/service/BerealTextTemplateService.java:19]: use Lists.newArrayList(iterator)

File: src/main/java/com/fbc/bot/config/annotation/AdminCheckAudit.java [hunk 6]
```diff
@@ -0,0 +1,11 @@
+package com.fbc.bot.config.annotation;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Target(ElementType.METHOD)
+@Retention(RetentionPolicy.RUNTIME)
+public @interface AdminCheckAudit {
```
Comment (vladilyushchenko) [inline src/main/java/com/fbc/bot/config/annotation/AdminCheckAudit.java:10]: renamte to AdminCheck

**Pull request merged**